lib/graphql/metrics/analyzer.rb

```diff
@@ -119,11 +133,13 @@ def extract_arguments(argument, field_defn, parent_input_object = nil)
       end
 
       def extract_argument(value, field_defn, parent_input_object = nil)
+        parent_field_type_name = field_defn.respond_to?(:owner) ? field_defn.owner.graphql_name : nil
```

Comment:
@eapache directive_definition worked perfectly, but with this caveat: 

`field_definition` and `directive_definition` will be different here (maybe I should change the name of variable?)

For `directive_definition` it will be class and for `field_definition` we will have instance here, looks like it's coming from here
https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/analysis/ast/visitor.rb#L103L130 (line 103 and 130)

`owner` is defined on instance in graphql-query gem for Field and for Directive

If we keep it as it is, should we return "Directive" instead of nil?

Reply:
I was thinking, maybe to keep compatibility, instead of renaming parent_field_name and parent_field_type_name we could add `parent_of_directive_name` and `parent_of_directive_type_name` and keep `parent_field_name` and `parent_field_type_name` with nil for directives (and  `parent_of_directive_name` and `parent_of_directive_type_name` set to nil in case if arguments belong to fields)

Reply:
Sketching another potential future state, with a [slightly modified Star Wars API example](https://graphql.org/learn/queries/#directives):

```graphql
query Hero($episode: Episode, $withFriends: Boolean!) {
  hero(episode: $episode) {
    name
    friends(first: 10) @include(if: $withFriends) { # 👀 
      name
    }
  }
}
```

For this query, looking only at the argument metrics we might gather for `friends(first: 10) @include(if: $withFriends)`

```ruby
# Metrics for hero.friends.first
static_metrics = {
  argument_name: "first",
  argument_type_name: "Integer",
  parent_name: "friends",
  parent_type_name: "Friend",
  parent_input_object_type: nil,
  default_used: false,
  value_is_null: false,
  value: 10,
}

# Metrics for hero.friends.@include.if
static_metrics = {
  argument_name: "if",
  argument_type_name: "Boolean",
  parent_name: "include",
  parent_type_name: "__Directive",
  parent_input_object_type: nil,
  default_used: false,
  value_is_null: false,
  value: false,
}
```

Perhaps just renaming to `parent_name` and `parent_type_name` with the convention of emitting reserved `__`-prefix type names for built-in types would be a reliable way for applications to distinguish directive arguments from field arguments. What do you both think?

Reply:
Thank you @chrisbutcher. 
I like idea having "__Directive", though not sure if it'll be enough to identify it (for example we could have @include applied to multiple fields)


Though looks like your example will provide different metrics for `hero.friends.first`. It will be 
```
static_metrics = {
  argument_name: "first",
  argument_type_name: "Integer",
  parent_name: "friends",
  parent_type_name: "QueryRoot",      #  <----- this
  parent_input_object_type: nil,
  default_used: false,
  value_is_null: false,
  value: 10,
}
```
and the reason is the way we handle arguments currently (i also found it confusing)
Here https://github.com/Shopify/graphql-metrics/blob/master/lib/graphql/metrics/analyzer.rb#L125l126 we have 

```ruby
 parent_field_name: field_defn.graphql_name,         
 parent_field_type_name: field_defn.owner.graphql_name,
```
from conversation with @eapache

>  The name of parent_field_type_name is a bit confusing but using owner is intentional. We wanted to be able to uniquely identify the specific field. There might be other fields on other types named post which return Post, but there's only ever one field named post on the specific type (QueryRoot)